docs/init.md

```diff
@@ -0,0 +1,201 @@
+
+3. No admin-specific components are loaded, which keeps front-end pages lean
+4. `WCS_Template_Loader` manages template overrides and customizations
+
+### Action Scheduler Events
+
+When an Action Scheduler event runs for subscriptions:
+
+1. WordPress core loads and the request is identified as a `cron` request (Action Scheduler uses WP Cron)
+2. The core initialization flow runs, but admin UI components are not loaded
+3. `WCS_Action_Scheduler` (initialized during core initialization) processes the scheduled event:
+   - Retrieves the correct subscription by ID from the action arguments
+   - Performs the scheduled action (payment processing, trial end, etc.)
+   - Updates subscription dates and statuses as appropriate
+
+4. The subscription status may change based on the event, which then triggers:
+   - Email notifications
+   - Payment processing
+   - Status changes and associated actions
+
+### REST API Requests
+
+For REST API requests related to subscriptions:
+
+1. WordPress identifies the request as a REST API request via `WC()->is_rest_api_request()`
+2. The core initialization flow runs
+3. WC Subscriptions checks if it's a REST API request using `wcs_is_rest_api_request()`
+4. Admin UI components are not loaded to keep the request lightweight
+5. Data handling remains consistent with standard requests, but presentation layers are skipped
+
+### AJAX Requests
+
+For AJAX requests related to subscriptions:
+
+1. WordPress identifies the request as an AJAX request via `wp_doing_ajax()`
+2. The core initialization flow runs
+3. WC Subscriptions checks if it's an AJAX request using `wcs_doing_ajax()`
+4. Depending on the specific AJAX action:
+   - Admin-specific components may be loaded for admin AJAX requests
+   - Front-end components for customer AJAX requests
+   - The response is typically JSON formatted data rather than rendered HTML
+
+## Key Differences Between Contexts
+
+| Context | Admin UI Components | Frontend Components | Template Loading | Performance Considerations |
+|---------|--------------------|--------------------|-----------------|---------------------------|
+| Admin Pages | ✅ All loaded | ❌ Not loaded | ✅ Admin templates only | Loads more components for full functionality |
+| Customer Pages | ❌ Not loaded | ✅ All loaded | ✅ Frontend templates | Focuses on cart, checkout, and my-account functionality |
+| Action Scheduler | ❌ Not loaded | ❌ Not loaded | ❌ Not needed | Lightweight, focuses on data processing only |
+| REST API | ❌ Not loaded | ⚠️ Partial loading | ❌ Not needed | Data-focused, returns structured data |
+| AJAX | ⚠️ Context-dependent | ⚠️ Context-dependent | ❌ Not needed | Minimal loading for fast responses |
+
+## Initialization Detection Functions
+
+WC Subscriptions provides helper functions to determine the current request context:
+
+```php
+// Check if current request is AJAX
+wcs_doing_ajax()
+
+// Check if current request is wp-cron (includes Action Scheduler)
```

Comment:
Similar to an earlier comment, this is misleading. All `wcs_doing_cron()` tells us, is that we are (or are not) inside the context of a cron request. I'd drop the Action Scheduler element.

Reply:
Yep, dropped AS element. However, it kind of sucks that we can't detect that and avoid loading bunch of unnecessary code like admin or store front stuff 😬